File: sdks/java/src/test/java/ai/nrouter/sdk/LiveTest.java
package ai.nrouter.sdk;

import static org.junit.jupiter.api.Assertions.*;

import java.util.List;
import java.util.Map;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;

class LiveTest {
    @Test
    @EnabledIfEnvironmentVariable(named = "NROUTER_LIVE", matches = "1")
    void liveClaudeRequestReturnsBillingMetadata() {
        String apiKey = System.getenv("NROUTER_API_KEY");
        String baseUrl = System.getenv().getOrDefault("NROUTER_BASE_URL", "http://127.0.0.1:4000/v1");
        assertNotNull(apiKey, "NROUTER_API_KEY is required for live tests");

        NRouterHttpResponse response = NRouter.httpClient(apiKey, baseUrl).messages(Map.of(
                "model", "claude-haiku-4-5-20251001",
                "max_tokens", 2,
                "messages", List.of(Map.of("role", "user", "content", "Reply OK"))));

        assertEquals(200, response.statusCode());
        assertNotNull(response.meta().requestId());
        assertTrue(response.meta().isPriced());
        assertNotNull(response.meta().cost());
        assertTrue(response.meta().cost() > 0.0);
    }
}

File: sdks/kotlin/src/test/kotlin/ai/nrouter/sdk/LiveTest.kt
package ai.nrouter.sdk

import kotlinx.coroutines.flow.toList
import kotlinx.coroutines.runBlocking
import kotlinx.coroutines.withTimeout
import org.json.JSONObject
import kotlin.test.Test
import kotlin.test.assertTrue

class LiveTest {
    @Test
    fun `live Claude stream reaches the configured gateway`() = runBlocking {
        if (System.getenv("NROUTER_LIVE") != "1") return@runBlocking
        val baseURL = System.getenv("NROUTER_BASE_URL") ?: NRouter.DEFAULT_BASE_URL
        val client = NRouter(baseURL = baseURL)
        val chunks = withTimeout(60_000) {
            client.messagesStream(
                JSONObject()
                    .put("model", "claude-haiku-4-5-20251001")
                    .put("max_tokens", 2)
                    .put(
                        "messages",
                        listOf(mapOf("role" to "user", "content" to "Reply OK")),
                    )
            ).toList()
        }
        assertTrue(chunks.any { it.delta.isNotEmpty() })
        assertTrue(!chunks.first().meta.requestId.isNullOrEmpty())
    }
}
